org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/GradleDistributionInfo.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core;
+
+import java.util.Optional;
+
+import org.eclipse.buildship.core.internal.DefaultGradleDistributionInfo;
+
+/**
+ * Represents a valid or invalid {@link GradleDistribution}.
+ *
+ * @author Donat Csikos
+ * @since 3.0
+ */
+public abstract class GradleDistributionInfo {
```

Comment:
❌ All of this is part of our preference infrastructure and not really interesting to other clients. We can cast to an internal interface to get the information we want. The public API should only have `GradleDistribution`